app/src/androidTest/java/ch/epfl/balelecbud/cloudMessaging/CloudMessagingServiceTest.java

```diff
@@ -0,0 +1,153 @@
+
+import android.content.Context;
+
+import androidx.test.ext.junit.runners.AndroidJUnit4;
+import androidx.test.rule.ActivityTestRule;
+import androidx.test.uiautomator.By;
+import androidx.test.uiautomator.UiDevice;
+import androidx.test.uiautomator.UiObject2;
+import androidx.test.uiautomator.Until;
+
+import com.google.firebase.messaging.RemoteMessage;
+
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import ch.epfl.balelecbud.BalelecbudApplication;
+import ch.epfl.balelecbud.WelcomeActivity;
+import ch.epfl.balelecbud.authentication.MockAuthenticator;
+import ch.epfl.balelecbud.models.User;
+import ch.epfl.balelecbud.notifications.NotificationMessage;
+import ch.epfl.balelecbud.util.database.MockDatabaseWrapper;
+import ch.epfl.balelecbud.util.http.HttpClient;
+import ch.epfl.balelecbud.util.http.MockHttpClient;
+
+import static androidx.test.platform.app.InstrumentationRegistry.getInstrumentation;
+import static org.hamcrest.Matchers.is;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertNull;
+import static org.junit.Assert.assertThat;
+
+@RunWith(AndroidJUnit4.class)
+public class CloudMessagingServiceTest {
+
+    public static final CloudMessagingService cloudMessagingService = new CloudMessagingService();
+
+    private final MockAuthenticator mockAuth = MockAuthenticator.getInstance();
+    private final MockDatabaseWrapper mockDB = MockDatabaseWrapper.getInstance();
+    private final HttpClient mockHttpClient = MockHttpClient.getInstance();
+    private final User user = MockDatabaseWrapper.celine;
+    private final String token = MockDatabaseWrapper.token;
+    private final String title = "This is a generic fun title!";
+    private final String body = "This is a fun text :)";
+
+    private UiDevice device;
+
+    @Rule
+    public final ActivityTestRule<WelcomeActivity> mActivityRule =
+            new ActivityTestRule<WelcomeActivity>(WelcomeActivity.class) {
+                @Override
+                protected void beforeActivityLaunched() {
+                    super.beforeActivityLaunched();
+                    BalelecbudApplication.setAppDatabaseWrapper(mockDB);
+                    BalelecbudApplication.setAppAuthenticator(mockAuth);
+                    BalelecbudApplication.setHttpClient(mockHttpClient);
+                    mockAuth.signOut();
+                    mockAuth.setCurrentUser(user);
+                    Message.setToken(token);
+                }
+            };
+
+    @Before
+    public void setup() {
+        cloudMessagingService.setContext(mActivityRule.getActivity());
+        device = UiDevice.getInstance(getInstrumentation());
+        if (device.hasObject(By.text("ALLOW"))) {
+            device.findObject(By.text("ALLOW")).click();
+            device.waitForWindowUpdate(null, 1_000);
+        }
```

Comment:
We don't need that now that the location switch has been moved to the setting activity